connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -359,7 +359,7 @@ private void convertMessages(ConsumerRecords<byte[], byte[]> msgs) {
                     new SinkRecord(msg.topic(), msg.partition(),
                             keyAndSchema.schema(), keyAndSchema.value(),
                             valueAndSchema.schema(), valueAndSchema.value(),
-                            msg.offset())
+                            msg.offset(), msg.timestamp(), msg.timestampType())
```

Comment:
Don't we need to be careful about the timestamp? I think since `ConsumerRecord` returns a `long` for this, couldn't it be the sentinel value `-1` if there is no timestamp? I think we need to check for `NO_TIMESTAMP` or if `timestampType()` is `NO_TIMESTAMP_TYPE`.